### PR TITLE
added UI for forgot-password step 1 2 and 3

### LINKED ISSUE #18: ForgetPassword Screen -> UI and Linking
**Is your feature request related to a problem? Please describe.**
Creating the UI of the screen after it is created by the design team and approved. Then linking the completed screen to the correct position in the widget tree of the app.

**Describe the solution you'd like**
1. Final UI creation as made and approved.
2. Linking of the completed screen at the correct position.

### PR DESCRIPTION
**Pull request by divy-arpit**
### Description

Added UI for the forgot password screen

**Type of Change:** (Delete irrelevant options)
- Code

- User Interface

**Code/Quality Assurance Only** (Delete irrelevant options)

- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

Tested on local machine.

### Checklist:

- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-review of my own code or materials
- [X] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] Any dependent changes have been merged
- [X] I have added one/more required dependencies
- [X] I have formatted the code using dart format (dartfmt)

**Code/Quality Assurance**
- [X] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

**Mention the dependencies/packages/plugins that have been added (if any)**

Android Emulator - Pixel_4_XL_API_30_5554 18-02-2021 12_47_28
Android Emulator - Pixel_4_XL_API_30_5554 18-02-2021 12_47_52

### CODE REVIEW

File: lib/screens/forgot_password/forgot_password.dart [hunk 1]
```diff
@@ -82,17 +242,217 @@ class ForgotPasswordScreen extends StatelessWidget {
...
+                        });
+                      },
+                      textColor: Colors.white,
+                      rightIcon: Icon(
+                        Icons.backspace,
+                        color: Colors.white,
+                      ),
+                      rightButtonFn: () {
+                        setState(() {
+                          otpEntered = otpEntered.substring(0, otpEntered.length - 1);
```
Comment (werainkhatri) [inline lib/screens/forgot_password/forgot_password.dart]: An error is thrown here (RangeError) when you click on right button (backspace) and otpEntered is empty. Add a check or something to prevent that.

File: lib/screens/forgot_password/forgot_password.dart [hunk 2]
```diff
@@ -106,17 +466,130 @@ class ForgotPasswordScreen extends StatelessWidget {
...
-    return Padding(
-      padding: const EdgeInsets.all(40.0),
-      child: Column(
+    double height = MediaQuery.of(context).size.height;
+    double bottom = MediaQuery.of(context).viewInsets.bottom;
+    double heightFactor = height / 1000;
+    if (_scrollController.hasClients) {
+      if (bottom > height * 0.25) {
+        _scrollController.animateTo(
+          bottom - height * 0.25,
```
Comment (werainkhatri) [inline lib/screens/forgot_password/forgot_password.dart]: this should be 0.5 instead of 0.25

**Review by werainkhatri — CHANGES_REQUESTED** [review]
1. reduce the size of Step 2/3 in the enter otp step
2. Add a resend button, add its fake functionality and center align the resend and verify buttons. refer the design. Add a timer as in the design. (Your can refer [this](https://pub.dev/packages/timer_button)) Discuss with the designer.
3. Add a back button, add the icon and change the title of the screen (to Verify your screen) according to the design.
4. Add horizontal padding to the fields in reset password step
5. Add the route to `main.dart` (in routes), its name to `strings.dart` and `login.dart` (navigator.of(context).pushNamed())

File: pubspec.lock [hunk 3]
```diff
@@ -163,6 +163,13 @@ packages:
       url: "https://pub.dartlang.org"
     source: hosted
     version: "0.16.1"
+  js:
```
Comment (Hard-Coder05) [inline pubspec.lock:166]: please revert the changes made in pubspe.lock
Also pull the latest code and remove the conflicts

**Review by Hard-Coder05 — CHANGES_REQUESTED** [review]
